### ISSUE TITLE
a quick way to change the units

### ISSUE THREAD

**Issue by alezzacreative** [GraphiteEditor/Graphite#586]
i would love us to have this very quick way to change the units in the ruler (its common in adobe products) an inkscape dev added it  just yesterday after alot of user requests.
https://user-images.githubusercontent.com/63913433/163684811-305787d7-88ce-4dc8-9410-42c14d19baa9.mp4
thank you

**Comment by Keavon** [comment]
This should be one of the document settings, which we'll need to make a menu for, to set the relationship between resolution and real-world units, as well as the default units to display. We should have a math-in-the-inputs-boxes system which does automatic unit conversions as well, so you can type any unit and it converts to dimensions then displays in the default display units.

**Comment by Keavon** [comment]
We should also have a way to let node parameters take a choice of multiple types of units. For example, the Resample Points node has Offset Start and Offset Stop parameters that currently accept a distance value. It would be good if we could let that node accept either a distance or a percentage, where the percentage is what factor along the curve's length to use as the start/stop offset. Percentage units should also work based on either a factor (`0-1`) or percent (`0%-100%`) as equivalent ways to express the same unit.

**Comment by ecstrema** [comment]
I would love to have that!

I wonder how common it is to mix units though. Should the whole document be in cm or px or whatever? Or should each individual value have a unit? Or maybe have a default unit, but make it modifiable on a case by case basis?

**Comment by Keavon** [comment]
A document will have a PPI value chosen by the user. It is a factor that relates physical units with pixel units.

**Comment by ecstrema** [comment]
> A document will have a PPI value chosen by the user. It is a factor that relates physical units with pixel units.

Well you see, PPI is pixel per inches I guess, but the point is that I guess I'd like to set my document to use centimeters, so I'd have to go and lookup cm to inches, then go into ppi, and write that number down as the ppi. 

It seems less user friendly than selecting: the units of this document is cm, and whenever you see 1, or 3.1 or whatever, you know these are centimeters.

Also, text has other units, so I wonder how much all of this should be general...

**Comment by Keavon** [comment]
Inches are exactly centimeters.

**Comment by ecstrema** [comment]
> Inches are exactly centimeters.

I probably don't understand what you are saying, but

<img width="697" height="285" alt="Image" src="https://github.com/user-attachments/assets/22480809-80e7-400b-93d7-a6e019f7157a" />

**Comment by Keavon** [comment]
Yes, the conversion factor is exact. All physical units are inherently metric.

**Comment by ecstrema** [comment]
Oh I get what you mean. Indeed, inches are based on metric units, but the opposite is not true.

So then, instead of setting the ppi, should it be something like this in the UI:

> One pixel accounts for exactly <number> <unit>

Where <number> is a number picker and <unit> is a unit picker?

**Comment by lumingzh** [comment]
"em" unit is also important for text typography, like [indentation](https://en.wikipedia.org/wiki/Indentation_(typesetting)).
